### ISSUE TITLE
Able to motion blur images that have transparency?

### ISSUE THREAD

**Issue by dene22** [fastred/MotionBlur#2]
![img_3968](https://cloud.githubusercontent.com/assets/2935071/5426395/15978118-8346-11e4-8e3a-60cc6b128c83.PNG)

I'm using a PNG with transparent and semi-transparent parts. These non-opaque parts become darkened when the motion blur is applied.

I hope this can be solved - it's very useful.

Best wishes,
Dene

**Comment by fastred** [comment]
I think the issue may lie in the PNG image. Could you try drawing the same letter in code, using Core Graphics?

**Comment by fastred** [comment]
Closing because of lack of response.

**Comment by sarperdag** [comment]
Hey @dene22 and @fastred, have you found any solution to this?

**Comment by fastred** [comment]
@sarperdag No, I didn't. Can you provide a sample project showing this error?

**Comment by sarperdag** [comment]
Here's a sample project showing the problem

https://dl.dropboxusercontent.com/u/1124427/TransparentMotionBlur.zip

**Comment by sarperdag** [comment]
@fastred @dene22 

Here's a sample project showing the problem

https://dl.dropboxusercontent.com/u/1124427/TransparentMotionBlur.zip

**Comment by fastred** [comment]
It looks like the issue lies in the filter: https://github.com/fastred/MotionBlur/blob/master/Classes/MotionBlurFilter.m

I don't have any quick ideas how to fix it, so I'll appreciate any help from you!

**Comment by sarperdag** [comment]
@fastred  Did you write the motion blur code yourself or got it from somewhere else?

**Comment by fastred** [comment]
The answer is on my blog:

> WWDC videos are always a good way to learn the basics of frameworks, so I started by watching sessions 514 – Advances in Core Image and 515 – Developing Core Image Filters for iOS from WWDC 2014. To my surprise, one of the presenters of session 515 showed how to write a motion blur filter. Let's see how it works.

http://holko.pl/2014/07/21/motion-blur/
